web/resources/js/transaction-form.js

```diff
@@ -63,4 +63,27 @@ $(function () {
     });
     // Show add button in last row
     split_rows.last().find(".split-add-button").removeClass("hidden");
+
+  document.querySelector("#accept_selected").addEventListener('click', function (element){
+      var ids = [];
+        $('#transactions_unconfirmed').bootstrapTable('getSelections').forEach(function (element){
+            ids.push(element["id"]);
+        })
+      if(ids.length > 0)
+      $.ajax({
+          type: "POST",
+          url: "/finance/transaction/confirm_selected",
+          data: {ids: ids},
+          success: function (result) {
+             console.log(result);
+          },
+          error:function (error){
+              console.log(error);
+          },
+          dataType: "json"
+        });
+      setTimeout(() => { window.location.reload(); }, 500);
```

Comment:
What's the precise workflow here? Why do you need a timer of `0.5s`? I was thinking you select the transactions and then press something. Perhaps i misunderstood the whole workflow.

Reply:
when reloading the page instantly the old transactions will appear again even tough they are confirmed maybe cause the query is not fast enough and I should just follow the the redirects but that did not worked some how. I try this with the other post method again without that

Reply:
I see, let me take a look.

Reply:
A clean way would be to wait for the request to finish by issuing a callback. That's precisely what the `fetch()` API is very good at, because it returns promises you can handle with `.then()`:
```js
      if (ids.length > 0) {
          fetch("/finance/transaction/confirm_selected", {
              method: "POST",
              body: JSON.stringify({ids: ids}),
              headers: { "Content-Type": "application/json" },
          })
              .then(r => {
                  console.debug("Got response, reloading page");
                  window.location.reload();
              })
              .catch(e => console.error(`Got error when submitting transactions to confirm: ${e}`));
 ```
 Haven't tested it, but that should be the Idea.

Reply:
when I used ajax I was not able to get a List when JSON.stringifying it i will test it

Reply:
Ah, right, you need to pass it as `formdata` if you access it via `request.form`. You will need to construct a `FormData` object and pass it in the `body` parameter of the `fetch` function (see [MDN](https://developer.mozilla.org/en-US/docs/Web/API/fetch#body))